### ISSUE TITLE
strToObjectRef doesn't fail gracefully

### ISSUE THREAD

**Issue by rerobins** [Linkurious/linkurious.js#372]
`strToObjectRef` implementation that is used in the legend and design plugin (if not others), doesn't fail gracefully if the string path doesn't exist in the object.  I would expect that if the search path doesn't exist it would return null instead of throwing an exception that is not caught any where.

I've replaced the implementation with `_.get()` from lodash as a placeholder.

A quick grep has shown that the following plugins also have an implementation of `strToObjectRef`.
* sigma.plugins.legend
* sigma.plugins.filter
* sigma.plugins.design
* sigma.exporters.xlsx
* sigma.exporters.spreadshhet
* sigma.exporters.graphml
* sigma.exporters.gexf

**Comment by sheymann** [comment]
Not sure that's a good thing to do. The information that the path exist should be known by your source code because it is bound to your data model. So calling this function with a bad path is a runtime error.

**Comment by rerobins** [comment]
I agree with everything you said. Except that execution completely fails. A better error message would he nice, but continue processing the rest of the nodes instead of exiting from the current processing.

If one node fails, the entire plugin shouldn't stop processing the rest of the nodes.

**Comment by sheymann** [comment]
I disagree, this is how you end up with hidden data flaws in production. Data integrity is paramount to data analysis, especially in networks where a single node can totally change the network structure.

**Comment by rerobins** [comment]
How about catching the exception to say datamodel failed at path x.y.z.a? Then rethrow?
